Add support for git creation date

timvink opened the pull request:
Closes #2551

jaceklaskowski commented on docs/setup/adding-a-git-repository.md:
```diff
@@ -186,6 +186,17 @@ The following options are supported:
 
+`enable_creation_date`{ #enable_creation_date }
+
+:   :octicons-milestone-24: Default: `false` – This option specifies whether to also add a page *Created* date at the bottom of each page.
```
nit: `also` seems redundant (think `This option` could be removed too)

timvink replied:
Fixed :)

jaceklaskowski commented on docs/setup/adding-a-git-repository.md:
```diff
@@ -123,7 +123,7 @@ The following options are supported:
 `enabled_if_env`{ #enabled_if_env }
 
-:   :octicons-milestone-24: Default: _none_ – This option defines whether the
+:   :octicons-milestone-24: Default: _none_ – Define whether the
```
Enables using data from git (commit?)...

jaceklaskowski commented on docs/setup/adding-a-git-repository.md:
```diff
@@ -164,7 +164,7 @@ The following options are supported:
 `type`{ #type }
 
-:   :octicons-milestone-24: Default: `date` – This option allows to change the
+:   :octicons-milestone-24: Default: `date` – Change the
```
The format of...

jaceklaskowski commented on docs/setup/adding-a-git-repository.md:
```diff
@@ -176,7 +176,7 @@ The following options are supported:
 `fallback_to_build_date`{ #fallback_to_build_date }
 
-:   :octicons-milestone-24: Default: `false` – This option specifies whether
+:   :octicons-milestone-24: Default: `false` – Specifies whether
```
Enables falling back to the time...

jaceklaskowski commented on docs/setup/adding-a-git-repository.md:
```diff
@@ -188,7 +188,7 @@ The following options are supported:
 `enable_creation_date`{ #enable_creation_date }
 
-:   :octicons-milestone-24: Default: `false` – This option specifies whether to also add a page *Created* date at the bottom of each page.
+:   :octicons-milestone-24: Default: `false` – Whether to add a *Created* date at the bottom of each page.
```
Enables adding...

jaceklaskowski reviewed (commented):
Sorry for nit-picking but spent too much time with doc writing and couldn't resist. HTH

timvink commented:
> Sorry for nit-picking but spent too much time with doc writing and couldn't resist. HTH

On the contrary, it shows you care. Much appreciated! Addressed the feedback in 33212c5

squidfunk reviewed (changes requested):
Thanks for taking the time to add support for this nice little plugin 🚀  Going down the path of adding support for more and more git-related information, I'd suggest that we settle on a better naming scheme. I suggest the following steps:

- Rename `source-date.html` into `source-file.html`(since this information is related to the current Markdown file). For downward compatibility until the next release, we keep `source-date.html` around and just include `source-meta.html` from there. In v8, we remove `source-date.html`

- Come up with a better naming for the translation keys. The current scheme can't be generalized nicely. My proposal:

  - `source.file.date.updated`
  - `source.file.date.created`
  - `source.file.author` (if support is added in the future)

  This would also be consistent with the new name of the partial. For downard compatibility, we can keep `source.revision.date` until v8.

timvink commented:
Nice improvement! Implemented in cee1eb4

squidfunk commented:
Thanks again! It'll be included in the next release.

squidfunk commented:
Trying to test it, I'm getting this error:

```
Traceback (most recent call last):
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/mkdocs/__main__.py", line 133, in serve_command
    serve.serve(
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 141, in serve
    config = builder()
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 136, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/mkdocs/commands/build.py", line 271, in build
    _populate_page(file.page, config, files, dirty)
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/mkdocs/commands/build.py", line 167, in _populate_page
    page.markdown = config['plugins'].run_event(
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/mkdocs/plugins.py", line 94, in run_event
    result = method(item, **kwargs)
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/mkdocs_git_revision_date_localized_plugin/plugin.py", line 179, in on_page_markdown
    commit_timestamp=self.util.get_git_commit_timestamp(
  File "/Users/squidfunk/Desktop/Projects/Personal/Sources/mkdocs-material-insiders/venv/lib/python3.8/site-packages/mkdocs_git_revision_date_localized_plugin/util.py", line 166, in get_git_commit_timestamp
    return int(commit_timestamp)
ValueError: invalid literal for int() with base 10: '1594984107\n1590932441'
```

Just added it on Material for MkDocs' own repository.

timvink commented:
Ah, I wouldn't have found that without running it on a large project. There are 2 creation dates: one when you created the `creating-your-site.md` in May 31st 2020, and another when you renamed the file (from docs/getting-started/creating-your-site.md → docs/creating-your-site.md ) on July 17th 2020.

Will make a fix and release soon.

timvink commented:
OK, I released a new version ([v0.9.2](https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/releases)) that addresses this.

I haven't tried building `material` with `npm` (not sure how). Would appreciate if you can test locally. I tried enabling the plugin with creation date enabled with the mkdocs.yml file currently on `squidfunk:master` and I get this error:

```
ERROR   -  Error building page 'changelog.md': source-file.html 

source-file.html
```

Any ideas?

Stanzilla commented:
@timvink try with https://github.com/squidfunk/mkdocs-material/pull/2627

timvink commented:
@Stanzilla Thnx, but that PR will break compatiblity:

> For downward compatibility until the next release, we keep source-date.html around and just include source-meta.html from there. In v8, we remove source-date.html (squidfunk)

That's why I went for this approach:

https://github.com/timvink/mkdocs-material/blob/cee1eb465d4a134d7f0e23f4a4885e02ecf70052/src/partials/source-date.html#L2-L3

timvink commented:
I changed

```
{% include "source-file.html" %} 
```

to 

```
{% include "partials/source-file.html" %} 
```

And that works for me. PR: https://github.com/squidfunk/mkdocs-material/pull/2629
